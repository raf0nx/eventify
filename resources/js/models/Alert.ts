import { EnumAlertType } from "@/enums/EnumAlertType";

export class AlertModel {
    showAlert = false;
    type = EnumAlertType.SUCCESS;
    message = "";
    dismissible = false;
    callback: Function | null = null;
    btnText: string = "";

    setShowAlert(showAlert: boolean): AlertModel {
        this.showAlert = showAlert;
        return this;
    }

    setType(type: EnumAlertType): AlertModel {
        this.type = type;
        return this;
    }

    setMessage(message: string): AlertModel {
        this.message = message;
        return this;
    }

    setDismissible(dismissible: boolean): AlertModel {
        this.dismissible = dismissible;
        return this;
    }

    setCallback(callback: Function): AlertModel {
        this.callback = callback;
        return this;
    }

    setBtnText(text: string): AlertModel {
        this.btnText = text;
        return this;
    }
}
