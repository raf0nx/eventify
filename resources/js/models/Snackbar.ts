import { EnumSnackbarColor } from "@/enums/EnumSnackbarColor";
import { EnumSnackbarIcon } from "@/enums/EnumSnackbarIcon";

export class SnackbarModel {
    showSnackbar = false;
    message = "";
    icon = EnumSnackbarIcon.SUCCESS;
    color = EnumSnackbarColor.SUCCESS;

    setShowSnackbar(value: boolean): SnackbarModel {
        this.showSnackbar = value;
        return this;
    }

    setMessage(message: string): SnackbarModel {
        this.message = message;
        return this;
    }

    setIcon(icon: EnumSnackbarIcon): SnackbarModel {
        this.icon = icon;
        return this;
    }

    setColor(color: EnumSnackbarColor): SnackbarModel {
        this.color = color;
        return this;
    }
}
