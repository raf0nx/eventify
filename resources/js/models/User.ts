export interface User {
    id: Number;
    name: String;
    email: string;
    created_at: Date;
    email_verified_at?: Date;
    updated_at: Date;
}
