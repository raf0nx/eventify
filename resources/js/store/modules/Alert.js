export const namespaced = true;

export const state = {
    message: null,
    showAlert: false
};

export const mutations = {
    SET_ALERT(state, message) {
        state.showAlert = true;
        state.message = message;
    }
};

export const actions = {};

export const getters = {
    message: state => {
        return state.message;
    },
    showAlert: state => {
        return state.showAlert;
    }
};
