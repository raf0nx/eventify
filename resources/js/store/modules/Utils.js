export const namespaced = true;

export const state = {
    showSnackbar: false,
    showAlert: false,
    loading: false
};
export const mutations = {
    SET_ALERT(state, setAlert) {
        state.showAlert = setAlert;
    },
    SET_LOADING(state, isLoading) {
        state.loading = isLoading;
    },
    SET_SNACKBAR(state, setSnackbar) {
        state.showSnackbar = setSnackbar;
    }
};
export const actions = {
    setSnackbar({ commit }) {
        commit("SET_SNACKBAR", true);
        setTimeout(() => commit("SET_SNACKBAR", false), 2500);
    }
};

export const getters = {
    showAlert: state => {
        return state.showAlert;
    },
    isLoading: state => {
        return state.loading;
    },
    showSnackbar: state => {
        return state.showSnackbar;
    }
};
