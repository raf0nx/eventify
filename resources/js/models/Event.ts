import { User } from "./User";

export interface Event {
    created_at: Date;
    description: string;
    event_image: string;
    id: number;
    name: string;
    start_datetime: Date;
    updated_at: Date;
    users: User[];
}
