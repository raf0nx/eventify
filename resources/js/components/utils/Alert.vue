<template>
	<v-alert
		v-if="alert.showAlert"
		class="mb-0 font-weight-bold text-center"
		data-cy="alert"
		border="bottom"
		tile
		dense
		:type="alert.type"
		:dismissible="alert.dismissible"
	>
		{{ alert.message }}
		<v-btn
			v-if="alert.callback"
			@click="alert.callback"
			outlined
			class="ml-4 font-weight-bold"
			small
		>
			{{ alert.btnText }}
		</v-btn>
	</v-alert>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import { UtilsModule } from "@modules/Utils";
	import { AlertModel } from "@/models/Alert";

	@Component
	export default class Alert extends Vue {
		get alert(): AlertModel {
			return UtilsModule.alert;
		}
	}
</script>

<style scoped>
</style>