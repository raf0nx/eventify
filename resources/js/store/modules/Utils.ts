import {
    Module,
    VuexModule,
    Mutation,
    Action,
    getModule
} from "vuex-module-decorators";

import store from "@/store/store";
import { SnackbarModel } from "@/models/Snackbar";
import { AlertModel } from "@/models/Alert";

export interface UtilsState {
    snackbarNotification: SnackbarModel;
    alertNotification: AlertModel;
    isLoading: boolean;
    navDrawer: boolean;
}

@Module({ dynamic: true, store, name: "utils" })
class Utils extends VuexModule implements UtilsState {
    snackbarNotification = new SnackbarModel();
    alertNotification = new AlertModel();
    isLoading = false;
    navDrawer = false;

    @Mutation
    private SET_ALERT(alert: AlertModel): void {
        this.alertNotification = alert;
    }

    @Mutation
    private SET_LOADER(isLoading: boolean): void {
        this.isLoading = isLoading;
    }

    @Mutation
    private SET_SNACKBAR(snackbar: SnackbarModel): void {
        this.snackbarNotification = snackbar;
    }

    @Mutation
    private SET_NAV_DRAWER(setNavDrawer: boolean): void {
        this.navDrawer = setNavDrawer;
    }

    @Action
    setAlert(alert: AlertModel): void {
        this.SET_ALERT(alert);
    }

    @Action
    setLoader(isLoading: boolean): void {
        this.SET_LOADER(isLoading);
    }

    @Action
    setSnackbar(snackbar: SnackbarModel): void {
        this.SET_SNACKBAR(snackbar);
    }

    @Action
    setNavDrawer(setNavDrawer: boolean): void {
        this.SET_NAV_DRAWER(setNavDrawer);
    }

    get alert(): AlertModel {
        return this.alertNotification;
    }

    get loader(): boolean {
        return this.isLoading;
    }

    get snackbar(): SnackbarModel {
        return this.snackbarNotification;
    }

    get isNavDrawer(): boolean {
        return this.navDrawer;
    }
}

export const UtilsModule = getModule(Utils);
