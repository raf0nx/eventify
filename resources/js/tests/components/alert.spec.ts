import Vuetify from "vuetify";
import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";

import Alert from "@components/utils/Alert.vue";
import { UtilsModule } from "@modules/Utils";
import { AlertModel } from "@/models/Alert";

const MESSAGE = "Message for test purposes";
const BTN_TEXT = "See more";

describe("Snackbar.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;
    let wrapper: Wrapper<Alert>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = shallowMount(Alert, {
            localVue,
            vuetify
        });

        UtilsModule.setAlert(
            new AlertModel()
                .setShowAlert(true)
                .setMessage(MESSAGE)
                .setCallback(() => true)
                .setBtnText(BTN_TEXT)
        );
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("Should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should get alert object from store", () => {
        // Arrange
        const alertModel = UtilsModule.alert;

        // Assert
        // @ts-ignore
        expect(wrapper.vm.alert).toEqual(alertModel);
    });
});
