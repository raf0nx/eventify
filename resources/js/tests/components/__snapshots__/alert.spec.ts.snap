// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Component Should match snapshot 1`] = `
"<v-alert-stub tile=\\"true\\" tag=\\"div\\" value=\\"true\\" border=\\"bottom\\" closelabel=\\"$vuetify.close\\" dense=\\"true\\" closeicon=\\"$cancel\\" icon=\\"\\" type=\\"success\\" data-cy=\\"alert\\" class=\\"mb-2 font-weight-bold d-flex justify-center\\">
  Message for test purposes
  <v-btn-stub outlined=\\"true\\" tag=\\"button\\" activeclass=\\"\\" small=\\"true\\" type=\\"button\\" class=\\"ml-4 font-weight-bold\\">
    See more
  </v-btn-stub>
</v-alert-stub>"
`;
